### DISCUSSION TITLE
Is there a roadmap?

### DISCUSSION THREAD

**Discussion by LegNeato** [immunant/c2rust#1038] [Q&A] [answered]
It is unclear to me as a potential contributor what the roadmap is and / or where I can jump in to help. Is there a table of C features with their `c2rust` status somewhere? Is the project focused on getting one particular C project working, or a bundle of them, or making working generated code better?

I don't expect a fully figured out plan, but most large projects have a status tracker / burn-down chart of some sort (examples:   https://github.com/uutils/coreutils#gnu-test-suite-compatibility, https://boajs.dev/boa/test262/, https://github.com/rust-lang/miri/issues/2057)

**Comment by thedataking** [comment, marked as answer]
Hi @LegNeato! Sorry, we don't maintain a roadmap of where the project is going; we're a tiny project compared to the ones you list. 

The bulk of our time is going towards analysis lifting the unsafe C that the transpiler produces into safe(r) Rust. That's not super easy to jump into as an external contributor. This means that the many serious papercuts with the transpiler have not received the attention they deserve. Nor has the refactoring component which is not currently operational; this prevents useful features such as making sure we only emit structure definitions once in the Rust.

There are several bugs where the transpiler generates incorrect code that we'd love to get help with. There are also larger defects such as the transpiler not currently preserving comments in the output Rust. I hope this helps a tiny bit, happy to drill down deeper if any of this sounds interesting to you.

**Comment by piaoger** [comment]
> There are also larger defects such as the transpiler not currently preserving comments in the output Rust.

@thedataking  I am trying to run c2rsut for a old c library which had already been transpiled before. But I am confused why all comments are there in the old one :)

**Comment by kkysen** [comment]
> I am trying to run c2rsut for a old c library which had already been transpiled before. But I am confused why all comments are there in the old one :)

An older version of `c2rust` used to preserve comments, but we unfortunately lost that when we moved from using the unstable `rustc` private crates to using `syn`, but it's harder to preserve comments the way `syn` is architected, and we unfortunately haven't had enough time to restore the comment preservation, though we'd very much like to.
